buildpacks/homebrew.go

```diff
@@ -106,38 +103,53 @@ func (bt HomebrewBuildTool) InstallDir() string {
 }
 
 func (bt HomebrewBuildTool) Install() error {
-	gi := goInfo.GetInfo()
+	if err := bt.installBrew(); err != nil {
+		return err
+	}
+	if !bt.IsPackage() {
+		return nil
+	}
+	if bt.PackageInstalled() {
+		log.Infof("Package %s already installed", bt.PackageVersionString())
+		return nil
+	}
+	log.Infof("Installing package %s", bt.PackageVersionString())
+	err := bt.installPackage()
+	if err != nil {
+		return fmt.Errorf("install homebrew: %w", err)
+	}
+	return nil
+}
 
-	var err error
-	switch gi.GoOS {
-	case "darwin":
-		err = bt.InstallDarwin()
-	case "linux":
-		err = bt.InstallLinux()
-	default:
-		err = fmt.Errorf("Unsupported platform: %s", gi.GoOS)
+func (bt HomebrewBuildTool) installBrew() error {
+	// If directory already exists, skip the install.
+	brewDir := bt.HomebrewDir()
+	if _, err := os.Stat(brewDir); err == nil {
+		log.Infof("brew installed in %s", brewDir)
+		return nil
 	}
 
+	log.Infof("Installing brew")
+	installDir := bt.InstallDir()
+	if err := os.MkdirAll(installDir, 0777); err != nil {
+		return fmt.Errorf("install homebrew: %w", err)
+	}
```

Comment:
Do we really need `0777`?

Reply:
So generally, unless there's a specific reason to restrict permissions, it is better to use `0666` or `0777` as modes to respect the user's [umask](https://en.wikipedia.org/wiki/Umask). `0022` is a fairly common umask value, which is why most programs automatically do "the right thing".

Reply:
Ah, didn't know about umask. Thanks for the link!